File: shared_resources/src/main/java/com/thewizrd/shared_resources/database/TzdbDAO.java
package com.thewizrd.shared_resources.database;

import androidx.room.Dao;
import androidx.room.Delete;
import androidx.room.Insert;
import androidx.room.OnConflictStrategy;
import androidx.room.Query;
import androidx.room.Transaction;

import com.thewizrd.shared_resources.tzdb.TZDB;

@Dao
public interface TzdbDAO {
    @Insert(onConflict = OnConflictStrategy.REPLACE)
    public void insertTZData(TZDB tzdb);

    @Delete
    public void deleteTZData(TZDB tzdb);

    @Transaction
    @Query("SELECT `tz_long` FROM tzdb WHERE `latitude` = :lat AND `longitude` = :lon")
    public String getTimeZoneData(double lat, double lon);
}

File: shared_resources/src/main/java/com/thewizrd/shared_resources/database/TzdbDAO.kt
package com.thewizrd.shared_resources.database

import androidx.room.*
import com.thewizrd.shared_resources.tzdb.TZDB

@Dao
interface TzdbDAO {
    @Insert(onConflict = OnConflictStrategy.REPLACE)
    suspend fun insertTZData(tzdb: TZDB)

    @Delete
    suspend fun deleteTZData(tzdb: TZDB)

    @Transaction
    @Query("SELECT `tz_long` FROM tzdb WHERE `latitude` = :lat AND `longitude` = :lon")
    suspend fun getTimeZoneData(lat: Double, lon: Double): String?
}